play-json/shared/src/main/scala/JsResult.scala

```diff
@@ -197,8 +197,33 @@ sealed trait JsResult[+A] { self =>
 }
 
 object JsResult {
+  import scala.util.{ Failure, Try, Success }
   import play.api.libs.functional._
 
+  case class Exception(cause: JsError)
+    extends IllegalArgumentException(Json stringify JsError.toJson(cause))
```

Comment:
Also, it might be nice to add a `toJson` method on the exception to make it easy to convert to a result:
```scala
lazy val toJson: JsValue = JsError.toJson(cause)
```
`exception.toJson` is a lot clearer than `JsError.toJson(exception.cause)`.